Integrations/integration-PagerDuty.yml

```diff
@@ -0,0 +1,475 @@
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(table_name, outputs, USERS_ON_CALL_NOW_HEADERS),
+            'EntryContext': {
+                'PagerDutyUser': context
+            }
+        }
+
+    def extract_incidents_data(incidents):
+        """Extact data about incidents."""
+        outputs = []
+        context = []
+        for i, incident in enumerate(incidents):
+            output = {}
+
+            output['ID'] = incident.get('id')
+            output['Status'] = incident.get('status')
+            output['Html Url'] = incident.get('html_url')
+            output['Created On'] = incident.get('created_at')
+            output['Incident Number'] = incident.get('incident_number')
+
+            output['Service ID'] = incident.get('service', {}).get('id')
+            output['Service Name'] = incident.get('service', {}).get('summary')
+            output['Service Html Url'] = incident.get('service', {}).get('html_url')
+
+
+            if len(incident.get('assignments', [])) > 0:
+                output['Assigned To User'] = incident['assignments'][0].get('assignee', {}).get('name')
+            else:
+                output['Assigned To User'] = '-'
+
+            output['Escalation Policy ID'] = incident.get('escalation_policy', {}).get('id')
+            output['Escalation Policy'] = incident.get('escalation_policy', {}).get('summary')
+
+            output['Trigger Type'] = incident.get('first_trigger_log_entry', {}).get('summary')
+            output['Trigger Details Html Url'] = incident.get('first_trigger_log_entry', {}).get('html_url')
+
+            output['Last Status Change On'] = incident.get('last_status_change_at')
+            output['Last Status Change By'] = incident.get('last_status_change_by', {}).get('summary')
+
+            output['Number Of Escalations'] = incident.get('number_of_escalations')
+            if output['Status'] == "resolved":
+                output['Resolved By User'] = output.get('Last Status Change By')
+            else:
+                output['Resolved By User'] = '-'
+
+            output['Assigned To'] = output['Assigned To User'] ###### This field is a duplicate, not sure why it's needed, should I delete?
+            output['Summary'] = incident.get('summary')
+
+            outputs.append(output)
+            context.append(output)
+
+        return {
+            'Type': entryTypes['note'],
+            'Contents': outputs,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(USERS_ON_CALL, outputs, INCIDENTS_HEADERS),
+            'EntryContext': {
+                'PagerDuty.Incidents': context
+            }
+        }
+
+    def extract_all_schedules_data(schedules):
+        """Extract the data about all the schedules."""
+        outputs = []
+        context = []
```

Comment:
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

Reply:
Done